Corrected terminology and updated a few things (#22972)
dotnet/docs: docs/framework/data/adonet/sql/sqlclient-support-for-high-availability-disaster-recovery.md

---
title: "SqlClient Support for High Availability, Disaster Recovery"
description: Learn about SqlClient application support for high-availability, disaster recovery in SQL Server, using Always On Availability Groups or Always On Failover Cluster Instances.
ms.date: "03/30/2017"
ms.assetid: 61e0b396-09d7-4e13-9711-7dcbcbd103a0
---
# SqlClient Support for High Availability, Disaster Recovery

> This topic discusses SqlClient support (added in .NET Framework 4.5) for high-availability, disaster recovery with the Always On features -- Always On Availability Groups (AGs) and Always On Failover Cluster Instances (FCIs) with SQL Server 2012 or later.  For more information about either Always On feature, see SQL Server Books Online.  

Reviewer: ```suggestion
This topic discusses SqlClient support (added in .NET Framework 4.5) for high-availability, disaster recovery with the Always On features -- Always On availability groups (AGs) and Always On failover cluster instances (FCIs) with SQL Server 2012 or later.  For more information about either Always On feature, see SQL Server Books Online.  
```

>  You can now specify ab availability group listener or the name of an FCI in the connection property. If a SqlClient application is connected to an AlwaysOn database that fails over, the original connection is broken and the application must open a new connection to continue work after the failover.  

Reviewer: ```suggestion
 You can now specify an availability group listener or the name of an FCI in the connection property. If a SqlClient application is connected to a database that fails over, the original connection is broken and the application must open a new connection to continue work after the failover.  
```

>  If you are not connecting to an AG or FCI, and if multiple IP addresses are associated with a hostname, SqlClient will iterate sequentially through all IP addresses associated with DNS entry. This can be time consuming if the first IP address returned by DNS server is not bound to any network interface card (NIC). When connecting to an AG's listener or an FCI, SqlClient attempts to establish connections to all IP addresses in parallel and if a connection attempt succeeds, the driver will discard any pending connection attempts.  

Reviewer: ```suggestion
 If you are not connecting to an AG or FCI, and if multiple IP addresses are associated with a hostname, SqlClient will iterate sequentially through all IP addresses associated with DNS entry. This can be time consuming if the first IP address returned by DNS server is not bound to any network interface card (NIC). When connecting an FCI, or to the listener of an availability group, SqlClient attempts to establish connections to all IP addresses in parallel and if a connection attempt succeeds, the driver discards any pending connection attempts.  
```

  
## Connecting With MultiSubnetFailover  

>  Always specify `MultiSubnetFailover=True` when connecting to the AG's listener or an FCI. `MultiSubnetFailover` enables faster failover for all AGs and or Failover Cluster Instance in SQL Server 2012 or later and will significantly reduce failover time for single and multi-subnet Always On topologies. During a multi-subnet failover, the client will attempt connections in parallel. During a subnet failover, will aggressively retry the TCP connection.  

Reviewer: ```suggestion
 Always specify `MultiSubnetFailover=True` when connecting to the FCI or the listener of an AG. `MultiSubnetFailover` enables faster failover for all AGs and or FCIs in SQL Server 2012 or later and significantly reduces failover time for single and multi-subnet Always On topologies. During a multi-subnet failover, the client attempts connections in parallel. During a subnet failover, the client aggressively retries the TCP connection.  
```
